src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_img2img.py

```diff
@@ -1315,14 +1315,14 @@ def __call__(
 
         # 5. Prepare timesteps
         def denoising_value_valid(dnv):
-            return isinstance(self.denoising_end, float) and 0 < dnv < 1
+            return isinstance(dnv, float) and 0 < dnv < 1
```

Comment:
@yiyixuxu Could you review please? The tests are now passing locally. I think the initial condition of checking `isinstance(self.denoising_end, float)` every time was incorrect because you can have denoising_end=None but denoising_start as not None.